File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java


class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Arrays.sort(nums1);
        Arrays.sort(nums2);
        int left = 0;
        int right = 0;
        ArrayList<Integer> output = new ArrayList<>();

        while (left < nums1.length && right < nums2.length) {
            if (nums1[left] == nums2[right]) {
                output.add(nums1[left]);
                left++;
                right++;
            } else if (nums1[left] < nums2[right]) {
                left++;
            } else {
                right++;
            }
        }

        
        HashSet<Integer> uniqueSet = new HashSet<>(output);

      
        int[] result = new int[uniqueSet.size()];
        int i = 0;
        for (int num : uniqueSet) {
            result[i++] = num;
        }

        return result;
    }
}

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection =(nums1, nums2) =>{
    nums1.sort();
    nums2.sort();
    const mySet1=new Set();
    for(let element of nums1){
        mySet1.add(element);
    }
    const intersectionset= new Set();
    for(let element of nums2){
        if(mySet1.has(element)){
            intersectionset.add(element);
        }
    }
    console.log(intersectionset);
    const arr= [...intersectionset];
    console.log(arr);

  return arr;
    
}